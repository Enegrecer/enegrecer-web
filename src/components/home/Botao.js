import React from 'react';
import PropTypes from 'prop-types';

const Botao = (props) => {
  function getClassName() {
    return `
      en-btn
      ${props.outline && 'en-btn--outline'}
      ${props.primary && 'en-btn--primary'}
      ${props.colored && 'en-btn--colored'}
    `;
  }

  return (
    <a
      onClick={props.onClick}
      className={getClassName(props)}
      role="button"
      tabIndex="0"
      name={props.name}
    >
      {props.children}
    </a>
  );
};

Botao.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.element,
  ]).isRequired,
  onClick: PropTypes.func.isRequired,
  outline: PropTypes.bool,
  primary: PropTypes.bool,
  colored: PropTypes.bool,
  name: PropTypes.string.isRequired,
};

Botao.defaultProps = {
  outline: false,
  primary: false,
  colored: false,
};

export default Botao;
