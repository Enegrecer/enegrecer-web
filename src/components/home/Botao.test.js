import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Botao from './Botao';

describe('Botao', () => {
  const onClickSpy = jest.fn();
  const wrapper = shallow(
    <Botao name="btn" onClick={onClickSpy}>
      Clique aqui
    </Botao>,
  );

  it('possui a mesma estrutura', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });

  it('tem a classe en-btn', () => {
    expect(wrapper.hasClass('en-btn')).toBeTruthy();
  });

  it('tem o texto "Clique aqui"', () => {
    expect(wrapper.text()).toEqual('Clique aqui');
  });

  it('tem o nome "btn"', () => {
    expect(wrapper.props().name).toEqual('btn');
  });

  it('n達o deve ter a classe en-btn--outline por padr達o', () => {
    expect(wrapper.hasClass('en-btn--outline')).not.toBeTruthy();
  });

  it('n達o deve ter a classe en-btn--primary por padr達o', () => {
    expect(wrapper.hasClass('en-btn--primary')).not.toBeTruthy();
  });

  describe('quando clica no botao', () => {
    beforeEach(() => {
      wrapper.simulate('click');
    });

    it('chama a funcao de onClick', () => {
      expect(onClickSpy).toBeCalled();
    });
  });

  describe('quando for outline', () => {
    beforeEach(() => {
      wrapper.setProps({ outline: true });
    });

    it('tem a classe en-btn--outline', () => {
      expect(wrapper.hasClass('en-btn--outline')).toBeTruthy();
    });
  });

  describe('quando for primary', () => {
    beforeEach(() => {
      wrapper.setProps({ primary: true });
    });

    it('tem a classe en-btn--primary', () => {
      expect(wrapper.hasClass('en-btn--primary')).toBeTruthy();
    });
  });

  describe('quando for colored', () => {
    beforeEach(() => {
      wrapper.setProps({ colored: true });
    });

    it('tem a classe en-btn--colored', () => {
      expect(wrapper.hasClass('en-btn--colored')).toBeTruthy();
    });
  });
});
