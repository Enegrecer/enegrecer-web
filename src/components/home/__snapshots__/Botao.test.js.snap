// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Botao possui a mesma estrutura 1`] = `
<a
  className="
      en-btn
      false
      false
      false
    "
  name="btn"
  onClick={[Function]}
  role="button"
  tabIndex="0"
>
  Clique aqui
</a>
`;
